DFE-Digital/apply-for-teacher-training
# Set up models and services for granular provider notification settings

## Context

As a provider user 
I need the ability to choose the type of email notification I receive
So that I only get notified about the things I need

## Changes proposed in this pull request

- add a feature flag
- add ProviderUserNotification model
- update NotificationsList service
- add a service to update all notification events
- update email sending services to account for different type of preferences

No UI changes.

## Guidance to review

Initially wanted to call the table 'provider_user_notification_preferences' but index name was too long.

To test:
- turn on the feature flag `FeatureFlag.activate(:configurable_provider_notifications)`
- create ProviderUserNotification  in the rails console and change it via provider interface and support interface
- check the values in the rails console

## Link to Trello card

https://trello.com/c/AtNkf165/3438-%F0%9F%8F%88-set-up-models-and-services-for-granular-provider-notification-settings

## Things to check

- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)
